Adding a distribution section to the JEP JEP and small tweaks

choldgraf opened the pull request:
Hey all - this PR builds off of #29 to add some language to the JEP enhancement proposal. There's a relevant issue to discuss this here: #27 

This PR is primarily about adding a "distribution" section to the "meta JEP". It mentions using a public website as an "archive" of finished JEPs. I tried not to tie it in to any specific technology, since I feel like that is more of an implementation detail and we can cross that bridge when we come to it. Hope that's OK.

The PR also:

* Cleans up a few formatting things I found in there, and adds links to the ipython/python *EP process
* Adds this JEP to the list of active JEPs in the readme (I am a bit unclear that status this JEP has...so I just left what was in the metadata for this one)

Lemme know what folks think!

willingc commented on 29-jep-process/README.md:
```diff
@@ -70,18 +70,18 @@ In order to transition out of the pre-proposal stage, the following checklist mu
-Outcome: 
+Outcome:
 
 The Shepard decides if the JEP criteria have been met.
```
```suggestion
The Sheperd decides if the JEP criteria have been met.
```

choldgraf replied:
lol shepherd is one of those words I have been misspelling for decades

willingc commented on 29-jep-process/README.md:
```diff
@@ -112,34 +112,49 @@ If in the course of the implementation, the implementer(s) can choose to withdra
+## Distribution
+
+This section describes how information about the JEP process (e.g., new JEPs, updates to
+current JEPs, etc) is communicated to the community.
```
```suggestion
current JEPs, etc) is communicated to the community.

Note: This JEP repo is the **canonical "source of truth"** for individual JEPs, the JEP process, and activity on JEPs. 
```

willingc commented on 29-jep-process/README.md:
```diff
@@ -112,34 +112,49 @@ If in the course of the implementation, the implementer(s) can choose to withdra
+
+### The JEP public archive website
+
+A list of all current JEPs and the content present in the pull request exists at
```
The JEP itself should be self contained for relevant links to discussions. At a minimum the website should have a reader-friendly version of JEPs.

I would clarify that the JEP itself, not the website, contains this information. The JEPs can be displayed to provide the community easier navigation and access to the information contained in the JEP repo.

willingc commented on 29-jep-process/README.md:
```diff
@@ -112,34 +112,49 @@ If in the course of the implementation, the implementer(s) can choose to withdra
+- **Jupyter Enhancement Proposal (JEP)**: A written document that describes a proposed unit of significant work on Jupyter.
 
-- Contributor: The person who is submitting the JEP, and might but not necessarily also have intention on organizing the implementation if so approved.
+- **Contributor**: The person who is submitting the JEP, and might but not necessarily also have intention on organizing the implementation if so approved.
```
```suggestion
- **Contributor**: The person (or persons) who is submitting the JEP. The implementation of a JEP may be done by others if so approved.
```

willingc commented on 29-jep-process/README.md:
```diff
@@ -112,34 +112,49 @@ If in the course of the implementation, the implementer(s) can choose to withdra
+- **Contributor**: The person who is submitting the JEP, and might but not necessarily also have intention on organizing the implementation if so approved.
 
-- Shepherd: A senior Jupyter contributor that manages the pre-proposal, submission, review, approval process of a JEP.
+- **Shepherd**: A senior Jupyter contributor that manages the pre-proposal, submission, review, approval process of a JEP.
```
```suggestion
- **Shepherd**: A senior Jupyter contributor who guides the **JEP Contributor** through the pre-proposal, submission, review, approval process of a JEP.
```

willingc commented on 29-jep-process/README.md:
```diff
@@ -112,34 +112,49 @@ If in the course of the implementation, the implementer(s) can choose to withdra
+- **Shepherd**: A senior Jupyter contributor that manages the pre-proposal, submission, review, approval process of a JEP.
 
-- Review Team (RT): A group of Jupyter contributors, with expertise in a particular area of the project that reviews and approves JEPs related to that area.
+- **Review Team (RT)**: A group of Jupyter contributors, with expertise in a particular area of the project that reviews and approves JEPs related to that area.
```
```suggestion
- **Review Team (RT)**: A group of Jupyter contributors, with expertise in a particular area of the project, that reviews JEPs related to that area.
```

willingc commented on 29-jep-process/README.md:
```diff
@@ -112,34 +112,49 @@ If in the course of the implementation, the implementer(s) can choose to withdra
+- **Review Team (RT)**: A group of Jupyter contributors, with expertise in a particular area of the project that reviews and approves JEPs related to that area.
 
-- Final Comment Period (FCP): A finite-time, post-approval period for the community to make final comments.
+- **Final Comment Period (FCP)**: A finite-time, post-approval period for the community to make final comments.
```
Shouldn't this be **pre-approval** so that folks have a window of time to comment prior to JEP acceptance.

willingc commented on README.md:
```diff
@@ -25,7 +25,7 @@ Below is a list of JEPs that have been submitted. To view the discussion around
 | 0026 | **Submitted** | [Add Language Server Support to Jupyter Server and jupyterlab-monaco](https://github.com/jupyter/enhancement-proposals/pull/26) |
 | 0028 | **Submitted** | [Standalone Jupyter Server](https://github.com/jupyter/enhancement-proposals/pull/28) |
-
+| 0029 | **Submitted** | [Jupyter Enhancement Proposal updates](https://github.com/jupyter/enhancement-proposals/pull/29)
```
Excellent!

captainsafia commented:
Looks good! I'll ditto @willingc's feedback.

Could you update the PR description to mention [the issue](https://github.com/jupyter/enhancement-proposals/issues/27) so that people reading the main issue can click off to get to the relevant PRs. Thanks!

choldgraf commented:
quick update:

* I added a sentence to linking to the issue at the top of this thread
* Suggested edits from @willingc accepted
* I changed "post-approval" to "pre-approval" since I agree this seems to make the most sense gauging from the flow of JEP states that @captainsafia described in the diagram.
* I tried updating the "distribution" section to be more clear about how the website text relates to the JEP content. Does that work @willingc ?

captainsafia commented:
cc: @mckev-amazon

willingc commented:
Thanks folks. 👍

choldgraf commented:
yay!
